### PR TITLE
FileManager: Add file properties tabs for ZIP and PDF files

### PR DESCRIPTION
**Pull request by AtkinsSJ**
image

image

A couple more file types you can now inspect. For ZIP, I left things so they could apply to other kinds of archive, we just don't have a convenient way to scan through them to get statistics yet.

### CODE REVIEW

File: Userland/Applications/FileManager/PropertiesWindowPDFTab.gml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+            @GUI::Label {
+                text: "Creator:"
+                text_alignment: "TopLeft"
+                fixed_width: 80
+            }
+
+            @GUI::Label {
+                name: "pdf_creator"
+                text: "SerenityOS Office"
```
Comment (trflynn89) [inline Userland/Applications/FileManager/PropertiesWindowPDFTab.gml:136]: Is there a benefit to having these non-real default text values? Seems like they could just be empty, then you don't have to go through to effort of setting them to an empty string if an error occurs

Comment (AtkinsSJ) [reply Userland/Applications/FileManager/PropertiesWindowPDFTab.gml:136]: Beyond making it a little nicer to work with in GML Playground, and my own amusement, not really. :sweat_smile:

File: Userland/Applications/FileManager/PropertiesWindow.cpp [hunk 2]
```diff
@@ -234,6 +241,39 @@ ErrorOr<void> PropertiesWindow::create_file_type_specific_tabs(GUI::TabWidget& t
...
+    if (!maybe_zip.has_value()) {
+        warnln("Failed to read zip file '{}' ", m_path);
+        return {};
+    }
+    auto zip = maybe_zip.release_value();
+
+    auto tab = TRY(tab_widget.try_add_tab<GUI::Widget>(TRY("Archive"_string)));
+    TRY(tab->load_from_gml(properties_window_archive_tab_gml));
+
+    size_t file_count = 0;
```
Comment (LucasChollet) [inline Userland/Applications/FileManager/PropertiesWindow.cpp:261]: That looks way too similar to what we have in `file.cpp` :eyes: 
What about putting that in a method of `Archive::Zip`?

Comment (vkoskiv) [reply Userland/Applications/FileManager/PropertiesWindow.cpp:261]: Yep, this same code is appearing in `file.cpp`, `unzip.cpp`, and now here. I think it makes sense to add an API to `LibArchive` to get this information. Doesn't have to be in this PR though, necessarily.

**Comment by AtkinsSJ** [comment]
Changes:
- Extract the logic for calculating total file count and size, as a `Archive::Statistics` class.
- Remove the dummy values from `PropertiesWindowPDFTab.gml` so we don't have to clear them. The dummy values are left in `PropertiesWindowArchiveTab.gml` and others where every value is always set anyway, since it does make editing the GML a little nicer.

**Pull request merged**